src/IndexBuilder.php

```diff
@@ -84,6 +84,10 @@ public function migrate(Index $currentIndex, array $params = [], array $context
             throw new RuntimeException(sprintf('Reindex call failed. %s', $response->getError()));
         }
 
+        if (!$waitForCompletion) {
```

Comment:
You might also call `$reindex->setWaitForCompletion()` as well.

This may push some additional change in this method. The `setWaitForCompletion()` could always receive `false`, then this method performs what it needs based on the `$waitForCompletion`. Or we also could call `setWaitForCompletion()` passing `$waitForCompletion` value and let the lib do the "wait" for us.